src/transport/MessageHeader.cpp

```diff
@@ -96,6 +97,7 @@ CHIP_ERROR MessageHeader::Decode(const uint8_t * data, size_t size, size_t * dec
     mSecureMsgType   = LittleEndian::Read16(p);
     mMessageId       = LittleEndian::Read32(p);
     mSecureSessionID = LittleEndian::Read32(p);
+    mProfileID       = LittleEndian::Read16(p);
     mTag             = LittleEndian::Read64(p);
 
     assert(p - data == kFixedHeaderSizeBytes);
```

Comment:
@shana-apple I think you need to update `kFixedHeaderSizeBytes` to account for 2 extra bytes of `ProfileID` at line 50.

Reply:
@shana-apple ?

Reply:
This should have been caught by the tests, let me take a look